### DISCUSSION TITLE
Read and Write Configs and save messages

### DISCUSSION THREAD

**Discussion by Kayto** [Kayto/Armachat-circuitpython#9] [General]
Currently reading from file and config.py works fine but write needs implementing - both to config.py and any save messages changes.

Circuit python write protects the drive unless boot.py allows it.

There is an option to save to text files, this may be better than using the config.py where config.py code may get broken by the user.
Example of writing to memory as text file here.

https://learn.adafruit.com/cpu-temperature-logging-with-circuit-python/logging-the-temperature

i.e something like user.txt to save username/contacts etc. 

             with open("/user.txt", "n") as fp:
                fp.write('{}\n'.format(user))
                fp.flush()

Reading as;
         with open("user.txt", "r") as f:
           user = f.readline()
           f.close()

**Comment by Kayto** [comment]
Save memory to flash does work if you change the file system to write mode by pressing the ALT key when booting and pressing the S key when viewing messages.

**Reply by richteel** [reply]
Yes, I put a progress bar in the boot.py file to give an indication of how much time you have to make a selection. After some thought, I think I should have the progress bar start full, rather than empty, then progress to empty. Don't know why I implemented it that way. I plan to change this evening.
